x-pack/plugin/core/src/main/java/org/elasticsearch/license/TransportGetLicenseAction.java

```diff
@@ -56,6 +60,10 @@ protected void masterOperation(
         ClusterState state,
         final ActionListener<GetLicenseResponse> listener
     ) throws ElasticsearchException {
-        listener.onResponse(new GetLicenseResponse(ClusterStateLicenseService.getLicense(state.metadata())));
+        if (licenseService instanceof ClusterStateLicenseService clusterStateLicenseService) {
+            listener.onResponse(new GetLicenseResponse(clusterStateLicenseService.getLicense(state.metadata())));
```

Comment:
`ClusterStateLicenseService` internally keeps a reference of `ClusterService`. Since this is already running on master node, I think calling `clusterStateLicenseService.getLicense()` directly should be equivalent if not better. We don't have to figure it out here since this is existing logic. But it feels like a tech debt.

Reply:
yeah .. it's probably safe to change, but I didn't want to introduce any potential functional changes.